lib/widgets/pill.dart

```diff
@@ -27,48 +26,18 @@ class Pill extends StatelessWidget {
   Widget build(BuildContext context) {
     final user = chatController.room?.getUser(identifier);
     final displayName = user?.displayName ?? identifier;
-    final avatarUrl =
-        user?.avatarUrl?.getDownloadLink(Matrix.of(context).client);
-    final avatarSize = DefaultTextStyle.of(context).style.fontSize ?? 14.0;
-    final padding = avatarSize / 20;
-    return InkWell(
-      child: Container(
-        padding: EdgeInsets.only(
-          top: padding,
-          bottom: padding,
-          left: avatarUrl != null ? padding * 3 : avatarSize / 2,
-          right: avatarSize / 2,
-        ),
-        decoration: BoxDecoration(
-          color: Colors.grey[800],
-          borderRadius: BorderRadius.all(Radius.circular(avatarSize + padding)),
-        ),
-        child: Row(
-          crossAxisAlignment: CrossAxisAlignment.center,
-          mainAxisSize: MainAxisSize.min,
-          children: <Widget>[
-            if (avatarUrl?.toString().isNotEmpty == true)
-              CircleAvatar(
-                radius: avatarSize / 2,
-                backgroundImage:
-                    CachedNetworkImageProvider(avatarUrl!.toString()),
-              ),
-            Flexible(
-              child: Text(
-                displayName.shortenDisplayName(
-                  maxCharacters: maxCharactersDisplayNameForPill,
-                ),
-                style: const TextStyle(color: Colors.white),
-                maxLines: 1,
-                overflow: TextOverflow.clip,
-              ),
-            ),
-          ],
-        ),
-      ),
+    return GestureDetector(
       onTap: () {
         onTap?.call(url);
       },
+      child: Text(
+        displayName.shortenDisplayName(
+          maxCharacters: maxCharactersDisplayNameForPill,
```

Comment:
If you didn’t add the person to your address book we display their name as they named themselves during sign-up (we limit the character count by 12 symbols excl. spaces.).

Pls update it